src/public/js/util.js

```diff
@@ -0,0 +1,218 @@
+     0x94, 0x8b, 0xc4, 0xd5, 0x9d, 0xf8, 0x90, 0x6b,
+     0xb1, 0x0d, 0xd6, 0xeb, 0xc6, 0x0e, 0xcf, 0xad,
+     0x08, 0x4e, 0xd7, 0xe3, 0x5d, 0x50, 0x1e, 0xb3,
+     0x5b, 0x23, 0x38, 0x34, 0x68, 0x46, 0x03, 0x8c,
+     0xdd, 0x9c, 0x7d, 0xa0, 0xcd, 0x1a, 0x41, 0x1c]);
+
+  function log3Util(x) {
+    return log3[x];
+  }
+
+  function exp3Util(x) {
+    return exp3[x];
+  }
+
+  function multInvUtil(x) {
+    return multInv[x];
+  }
+
+  function fieldDivide(divisor, dividend) {
+    var t = 0;
+    if (divisor == 0 || dividend == 0) {
+      return 0;
+    }
+    t = log3[dividend] - log3[divisor];
+    if (t < 0) {
+      t = t + 255;
+    }
+    return exp3[t];
+  }
+
+  function fieldMultiply(mult1, mult2) {
+    var t = 0;
+    if (mult1 == 0 || mult2 == 0) {
+      return 0;
+    }
+    t = log3[mult1] + log3[mult2];
+    if (t > 255) {
+      t = t - 255;
+    }
+    return exp3[t];
+  }
+
+  function shift(msg, rot) {
+    return msg.split('').reduce(function(accumulator, letter, index) {
+      var shiftedCharCode = letter.charCodeAt() + rot;
+      var shiftedLetter = String.fromCharCode(shiftedCharCode);
+      return accumulator + shiftedLetter;
+    }, '');
+  }
+
+  function polynomialDiv(dividend, divisor, n, k) {
+    if (divisor.length != k || dividend.length != n || k >= n) {
+      return null;
+    }
+    // Precompute divisor's highest order inverse and distribute
+    // to all of divisor
+    var invVal = multInv[divisor[0]];
+    for (i = 0; i < k; i++) {
+      divisor[i] = fieldMultiply(divisor[i], invVal);
+    }
+    for (i = 0; i < n - k; i++) {
+
+      // Find how many of highest order of scaled divisor go in to
+      // current highest order of leftover dividend
+      // Because of precompute it is simply the coefficient of the
+      // highest order of dividend
+      var quotient = dividend[i];
+
+      // Find scaled divisor to subtract from current remainder of dividend
+      // by multiplying each coefficient of precomputed divisor by current
+      // highest order quotient
+      var scaledDivisor = new Uint8Array(k);
+      for (j = 0; j < k; j++) {
+        scaledDivisor[j] = fieldMultiply(divisor[j], quotient);
+      }
+
+      // Subtract scaled divisor from current remainder of dividend
+      // to get new remainder of dividend which will be 1 order less
+      for (j = i; j < i + k; j++) {
+        // Galois extension field subtraction is xor with base field 2
+        dividend[j] = dividend[j] ^ scaledDivisor[j - i];
+      }
+    }
+
+    // I am sure there is a better way to do this but I am bad.
+    var remainder = new Uint8Array(n - k - 1);
+    for (i = 0; i < n - k - 1; i++) {
+      remainder[i] = dividend[i + k + 1];
+    }
+    // Adding this as kinda failsafe test case
+    for (i = 0; i < k + 1; i++) {
+      if (dividend[i] != 0) {
+        return null;
+      }
+    }
+    return remainder;
+  }
+
+  return {
+    log3: log3Util,
+    exp3: exp3Util,
+    multInv: multInvUtil,
```

Comment:
part of the appeal of the IIFE is that I don't think we ever have to access log3, exp3, multInv directly -- I would keep them private instead of making them "public" by including them in the return object